WorkspaceServer/Kernel/CodeSubmissionProcessors.cs

```diff
@@ -21,9 +21,10 @@ public class CodeSubmissionProcessors
         public CodeSubmissionProcessors()
         {
             _rootCommand = new RootCommand();
+            _parser = new CommandLineBuilder(_rootCommand).Build();
         }
 
-        public void Register(ICodeSubmissionProcessor processor)
+        public void Add(ICodeSubmissionProcessor processor)
         {
             _processors[processor.Command] = processor;
```

Comment:
`Add` seems like a misleading name if this can also replace an existing processor.

Reply:
changed ot add so we throw exceptions when overriding

Reply:
This still obscure the intention. A more honest name would be something like SetProcessorForCommand.

---
In reply to: [301570041](https://github.com/dotnet/try/pull/313#discussion_r301570041) [](ancestors = 301570041)